src-graphql/product-query.js

```diff
@@ -0,0 +1,39 @@
+import optionQuery from './option-query';
+import imagesQuery from './images-query';
+import variantQuery from './variant-query';
+import createGid from './create-gid';
+
+export default function productQuery({client, id}, fields = ['id', 'createdAt', 'updatedAt', 'bodyHtml', 'handle', 'productType',
+  'title', 'vendor', 'tags', 'publishedAt', ['options', optionQuery()], ['images', imagesQuery()], ['variants', variantQuery()]]) {
+  let query;
```

Comment:
You don't need a `let` query. You can do an `if (id) { return client.query( ... ) } else { return client.query( ... ) }`